node/pkg/processor/broadcast.go

```diff
@@ -72,21 +85,14 @@ func (p *Processor) broadcastSignature(
 	}
 
 	p.state.signatures[hash].ourObservation = o
+	p.state.signatures[hash].ourObs = ourObs
 	p.state.signatures[hash].ourMsg = msg
 	p.state.signatures[hash].txHash = txhash
 	p.state.signatures[hash].source = o.GetEmitterChain().String()
 	p.state.signatures[hash].gs = p.gs // guaranteed to match ourObservation - there's no concurrent access to p.gs
 
-	// Fast path for our own signature
-	// send to obsvC directly if there is capacity, otherwise do it in a go routine.
-	// We can't block here because the same process would be responsible for reading from obsvC.
-	om := node_common.CreateMsgWithTimestamp[gossipv1.SignedObservation](&obsv)
-	select {
-	case p.obsvC <- om:
-	default:
-		go func() { p.obsvC <- om }()
-	}
-
+	// Post the signature to ourselves.
+	p.handleSingleObservation(addr, ourObs)
```

Comment:
My initial guess is that, since the same process is responsible for this anyway, it could save resources handling this directly, but at the risk of expedited treatment for our own observations. That may not necessarily be a bad thing though, since publishing consensus currently requires our own observation.

Reply:
It seems reasonable to me to give this a shot with the aim to further optimize it, particularly with #3988 on the horizon, lessening the immediate impact of this call. I always thought it was odd that the injection was handled in `broadcastSignature` anyway when `HandleQuorum` handles its own state setting and only uses `broadcastSignedVAA` to perform the broadcast.